pgmpy/inference/base.py

```diff
@@ -79,3 +79,47 @@ def __init__(self, model):
             for factor in model.get_factors():
                 for var in factor.variables:
                     self.factors[var].append(factor)
+
+
+class Approx_Inference:
+    """
+    Base class for the approximate inference algorithm.
+
+    Parameters
+    ----------
+    model: pgmpy.models.MarkovModel
+    We need a pairwise MarkovModel for Approx. inference.
+    """
+    def __init__(self, model):
+
+        # List of the variables in the present pairwise Markov Model
+        self.variables = []
+
+        # List of the cardinalities of the above variables in that order
+        self.cardinality = []
+
+        # List of all the factors that are present in the current model
+        self.factors = []
+
+        # List of all the factors values that are present in the current model
+        self.factor_values = []
+
+        # The integrality gap achieving which the approximation is considered completed.
+        self.int_gap = 0.002
+
+        # Number of iterations that the approximation algorithm should run, if say int_gap is not achieved
+        self.niter = 1000
```

Comment:
```n_iter``` is a misnomer. Instead rename it to ```n_max_iter```. Furthermore, there are some approx inference algorithms that are not going to run for this huge number of iterations. You can keep it as 100 in general, and for your specific algorithm if you think 1000 is a good number change it 1000.